frontend/src/CompanyInfo.jsx

```diff
@@ -101,11 +101,20 @@ const AddToPortfolio = ({ companyId }) => {
               );
             })}
           {portfoliosToAddToo.length === 0 && (
-            <p className="text-sm font-bold text-center">
-              {" "}
-              either no selected portfolios or this company is already in all
-              portfolios
-            </p>
+            <>
+              <p className="text-sm font-bold text-center">
+                {" "}
+                either no selected portfolios or this company is already in all
```

Comment:
unrelated nit: this phrasing could be punchier. Maybe something like "No valid portfolios selected"

Even better, maybe there should be some visual indicator that the company is already in a given portfolio? You could even go as far as not displaying portfolios that already contain the company.